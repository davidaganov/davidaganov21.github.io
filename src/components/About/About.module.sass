.about
  padding-top: 3rem

.content
  display: flex
  justify-content: space-between
  margin-top: 3rem
  @media (max-width: 999px)
    flex-direction: column-reverse

.left
  color: var(--color-white-700)
  p:not(:last-of-type)
    margin-bottom: 2rem
  strong
    color: var(--color-accent)
  @media (min-width: 1000px)
    width: 54%
    .picture
      display: none
  @media (max-width: 999px) and (min-width: 576px)
    max-width: 95%
  @media (max-width: 575px)
    display: flex
    flex-direction: column
    align-items: center

.btn
  margin-top: 4rem

.right
  @media (max-width: 999px)
    display: none

.picture
  @media (min-width: 1000px)
    margin-top: 2rem
    width: 35rem
  @media (max-width: 999px) and (min-width: 576px)
    max-width: 13rem
    max-height: 13rem
    margin: .7rem 2rem 1rem 0
    float: left
  @media (max-width: 575px)
    width: 10rem
    height: 10rem
    margin-bottom: 2rem
