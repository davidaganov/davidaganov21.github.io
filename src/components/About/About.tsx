import { useTranslation } from "react-i18next"
import styles from "./About.module.sass"

import { Title, DownloadResume } from "../"

export const About = () => {
  const { t } = useTranslation()

  const renderItems = () => {
    const descripton = Object.keys(t("about.description", { returnObjects: true }))
    const items = descripton.map((i) => {
      return (
        <p
          key={i}
          dangerouslySetInnerHTML={{ __html: t(`about.description.${i}`) }}
        />
      )
    })

    return <>{items}</>
  }

  const items = renderItems()

  return (
    <section
      className={styles.about}
      id="about"
    >
      <div className="inner">
        <Title
          link="#about"
          title={t("about.title")}
        />

        <div className={styles.content}>
          <div className={styles.left}>
            <img
              className={styles.picture}
              src={require("../../assets/images/avatar.png")}
              alt="My avatar"
              width="350"
              height="350"
            />
            {items}
            <DownloadResume />
          </div>
          <div className={styles.right}>
            <img
              className={styles.picture}
              src={require("../../assets/images/avatar.png")}
              alt="My avatar"
              width="350"
              height="350"
            />
          </div>
        </div>
      </div>
    </section>
  )
}
