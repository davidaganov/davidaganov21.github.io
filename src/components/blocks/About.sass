.about
  &__content
    display: flex
    justify-content: space-between
    margin-top: 3rem

  &__left
    width: 50%
    color: var(--color-white-700)
    p:not(:last-of-type)
      margin-bottom: 2rem

  &__picture
    position: relative
    width: 30rem
    height: 30rem
    background-color: var(--color-accent)
    &::after
      position: absolute
      content: ""
      width: 100%
      height: 100%
      border: .2rem solid var(--color-accent)
      right: -1rem
      top: -1rem
