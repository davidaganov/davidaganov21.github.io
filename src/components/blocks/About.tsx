import Title from "../base/Title"
import data from "../../data"

export default function About() {
  const renderItems = () => {
    const items = data.about.map((item, i) => {
      return (
        <p
          key={i}
          dangerouslySetInnerHTML={{ __html: item }}
        />
      )
    })

    return <>{items}</>
  }

  const items = renderItems()

  return (
    <section
      className="about"
      id="about"
    >
      <div className="about__inner inner">
        <Title
          classes="about__title"
          number={1}
          link="#about"
          title="About Me"
        />

        <div className="about__content">
          <div className="about__left">
            <img
              className="about__picture"
              src={require("../../assets/images/avatar.png")}
              alt="My avatar"
              width="350"
              height="350"
            />
            {items}
          </div>
          <div className="about__right">
            <img
              className="about__picture"
              src={require("../../assets/images/avatar.png")}
              alt=""
            />
          </div>
        </div>
      </div>
    </section>
  )
}
