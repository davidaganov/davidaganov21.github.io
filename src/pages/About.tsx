import { useTranslation } from "react-i18next"
import { Navbar, Footer } from "../layout"
import { Links } from "../components"

export default function About() {
  const { t } = useTranslation()

  return (
    <div className="wrapper">
      <Navbar
        single={true}
        list={t("navbar.about", { returnObjects: true })}
      />
      <main>
        <Links />
      </main>
      <Footer />
    </div>
  )
}
