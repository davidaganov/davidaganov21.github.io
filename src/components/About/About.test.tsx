import "@testing-library/jest-dom"
import { render, screen } from "@testing-library/react"
import { About } from "./About"

describe("About component", () => {
  it("About renders", () => {
    render(<About />)

    expect(screen.getByText("#").closest("a")).toHaveAttribute("href", "#about")
  })

  it("About snapshot", () => {
    const about = render(<About />)

    expect(about).toMatchSnapshot()
  })
})

// import "@testing-library/jest-dom"
// import { render, screen } from "@testing-library/react"
// import { About } from "./About"

// describe("About component", () => {
//   it("About renders", () => {
//     render(<About />)

//     // expect(screen.getByText(/test/i)).toBeInTheDocument()
//   })

//   it("About snapshot", () => {
//     const about = render(<About />)

//     expect(about).toMatchSnapshot()
//   })
// })
