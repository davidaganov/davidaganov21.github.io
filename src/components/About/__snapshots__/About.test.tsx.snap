// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`About component About snapshot 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <section
        class="about"
        id="about"
      >
        <div
          class="inner"
        >
          <div
            class="title ltr"
          >
            <h2>
              <span
                aria-hidden="true"
              >
                0
                1
                .
              </span>
               
              About me
            </h2>
            <a
              aria-label="About me"
              href="#about"
            >
              #
            </a>
          </div>
          <div
            class="content"
          >
            <div
              class="left"
            >
              <img
                alt="My avatar"
                class="picture"
                height="350"
                src="avatar.png"
                width="350"
              />
              <p>
                I started getting interested in web development back in 2016, when I watched a lot of YouTube videos about development, looked for free courses on the Internet and tried to create my own sites.
              </p>
              <p>
                Years later and after completing my studies, I was lucky to connect my life with my hobby and get a job as a frontend developer in a company developing mobile games and individual projects.
              </p>
              <p>
                During the year and a half spent in this company, I learned a lot. My main activity was development of products with 
                <strong>
                  Nuxt.js
                </strong>
                 framework and creation of many landing pages.
              </p>
              <p>
                The company also gave me the opportunity to take courses on 
                <strong>
                  React
                </strong>
                 and 
                <strong>
                  React Native
                </strong>
                 development.
              </p>
            </div>
            <div
              class="right"
            >
              <img
                alt="My avatar"
                class="picture"
                height="350"
                src="avatar.png"
                width="350"
              />
            </div>
          </div>
        </div>
      </section>
    </div>
  </body>,
  "container": <div>
    <section
      class="about"
      id="about"
    >
      <div
        class="inner"
      >
        <div
          class="title ltr"
        >
          <h2>
            <span
              aria-hidden="true"
            >
              0
              1
              .
            </span>
             
            About me
          </h2>
          <a
            aria-label="About me"
            href="#about"
          >
            #
          </a>
        </div>
        <div
          class="content"
        >
          <div
            class="left"
          >
            <img
              alt="My avatar"
              class="picture"
              height="350"
              src="avatar.png"
              width="350"
            />
            <p>
              I started getting interested in web development back in 2016, when I watched a lot of YouTube videos about development, looked for free courses on the Internet and tried to create my own sites.
            </p>
            <p>
              Years later and after completing my studies, I was lucky to connect my life with my hobby and get a job as a frontend developer in a company developing mobile games and individual projects.
            </p>
            <p>
              During the year and a half spent in this company, I learned a lot. My main activity was development of products with 
              <strong>
                Nuxt.js
              </strong>
               framework and creation of many landing pages.
            </p>
            <p>
              The company also gave me the opportunity to take courses on 
              <strong>
                React
              </strong>
               and 
              <strong>
                React Native
              </strong>
               development.
            </p>
          </div>
          <div
            class="right"
          >
            <img
              alt="My avatar"
              class="picture"
              height="350"
              src="avatar.png"
              width="350"
            />
          </div>
        </div>
      </div>
    </section>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
